### ISSUE TITLE
test: add 13 sampler unit tests covering all sampling parameters and boundary conditions

### ISSUE THREAD

**Issue by devswha** [devswha/zipcode#62]
## Summary

Added 13 deterministic unit tests to `crates/inference/src/sampler.rs`, increasing coverage from 1 → 14 tests. The sampler is a critical inference component (temperature, top-k, top-p, repeat penalty, softmax) that previously had only a single greedy-sampling test.

### Greedy / temperature=0
- `test_greedy_with_equal_logits_picks_last_max` — documents `Iterator::max_by` tie-breaking behavior (returns last max, not first)

### Temperature scaling
- `test_temperature_scaling_preserves_argmax_for_extreme_values` — near-zero temp (0.001) still picks the dominant logit

### Repeat penalty (5 tests)
- `test_repeat_penalty_divides_positive_logits` — penalty > 1 divides positive logits, changing the argmax
- `test_repeat_penalty_multiplies_negative_logits` — penalty multiplies negative logits (pushes further negative)
- `test_repeat_penalty_window_respects_repeat_last_n` — tokens outside the `repeat_last_n` window are not penalized
- `test_repeat_penalty_no_op_when_penalty_is_one` — penalty=1.0 is a no-op

### Top-k filtering (2 tests)
- `test_top_k_filters_to_k_highest` — top_k=1 collapses to greedy
- `test_top_k_disabled_when_zero` — top_k=0 passes all tokens through

### Top-p (nucleus) filtering (2 tests)
- `test_top_p_one_point_zero_disables_filtering` — top_p=1.0 is a no-op
- `test_top_p_near_zero_concentrates_on_highest` — extreme top_p isolates the single highest-probability token

### Boundary conditions (2 tests)
- `test_single_logit_always_returns_zero` — single-element tensor always returns index 0
- `test_all_probs_zero_returns_token_zero` — sum=0 fallback returns token 0

### Integration
- `test_repeat_penalty_and_greedy_combined` — repeat penalty + greedy interact correctly

## Verification

```
cargo test -p zipcode-inference sampler  → 14 passed, 0 failed
cargo test --workspace                     → 80 passed, 0 failed
cargo clippy --workspace --all-targets     → 0 warnings
cargo fmt --all -- --check                 → clean
```

## Commit

`f22fee8` on `main`

## Notes

- All tests are deterministic (greedy/temp=0.001) — no flaky random assertions
- `omx exec --madmax` was attempted but failed with 401 Unauthorized; tests were written directly using local tools
- Discovered and documented that greedy tie-breaking picks the **last** max via `Iterator::max_by`, which is not stable — this is an existing behavior worth being aware of

**Comment by devswha** [comment]
Resolved by f22fee8 on main. Issue title matched the commit subject; close-on-merge wasn't wired so the tracking issue stayed open.
